jgrapht-core/src/test/java/org/jgrapht/alg/matching/blossom/v5/KolmogorovWeightedMatchingTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * (C) Copyright 2018-2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.matching.blossom.v5;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.interfaces.MatchingAlgorithm;
+import org.jgrapht.graph.DefaultUndirectedWeightedGraph;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.util.SupplierUtil;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static junit.framework.TestCase.*;
+import static org.jgrapht.alg.matching.blossom.v5.KolmogorovWeightedPerfectMatching.EPS;
+
+/**
+ * Unit tests for the {@link KolmogorovWeightedMatching}
+ *
+ * @author Timofey Chudakov
+ */
+@RunWith(Parameterized.class)
+
+public class KolmogorovWeightedMatchingTest {
+    /**
+     * Algorithm options
+     */
+    private BlossomVOptions options;
+    /**
+     * Algorithm goal
+     */
+    private boolean maximize;
+
+    public KolmogorovWeightedMatchingTest(BlossomVOptions options, boolean maximize) {
+        this.options = options;
+        this.maximize = maximize;
+    }
+
+    @Parameterized.Parameters
+    public static Object[][] params() {
+        BlossomVOptions[] options = BlossomVOptions.ALL_OPTIONS;
+        Object[][] params = new Object[2 * options.length][2];
+        for (int i = 0; i < options.length; i++) {
```

Comment:
Euhm, could you add some comments here describing what is is that you are doing here? :)